api/client/create.go

```diff
@@ -77,7 +77,7 @@ func newCIDFile(path string) (*cidFile, error) {
 	return &cidFile{path: path, file: f}, nil
 }
 
-func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runconfig.HostConfig, cidfile, name string) (*types.ContainerCreateResponse, error) {
+func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runconfig.HostConfig, cidfile, name string, pull bool) (*types.ContainerCreateResponse, error) {
```

Comment:
I think @cpuguy83 means a struct to pass options to `createContainer()` in stead of passing a long list of arguments. That struct is just for internal use, not for the API, IIRC that approach is used in  other locations